### PR TITLE
[APM-297894] Improve diagnostics of GCP metric ingest function

### PR DESCRIPTION
**Pull request by dtwwolinski**
- moved fast check dynatrace to print token metadata and URL during execution.

### CODE REVIEW

File: src/lib/fast_check.py [hunk 1]
```diff
@@ -96,6 +96,25 @@ async def get_dynatrace_token_metadata(dt_session: ClientSession, context: Loggi
...
 
+async def check_dynatrace(logging_context: LoggingContext, project_id, dt_session: ClientSession, dynatrace_url, dynatrace_access_key):
+    try:
+
+        if not dynatrace_url or not dynatrace_access_key:
+            logging_context.log(f'ERROR No Dynatrace secrets: DYNATRACE_URL, DYNATRACE_ACCESS_KEY for project: {project_id}.'
+                                     f'Add required secrets to Secret Manager.')
+            return None
+        logging_context.log(f"Using [DYNATRACE_URL] Dynatrace endpoint: {dynatrace_url}")
+        logging_context.log(f'Using [DYNATRACE_ACCESS_KEY]: {dynatrace_access_key[0:dynatrace_access_key.rfind(".")]}.')
```
Comment (sq2gxo) [inline src/lib/fast_check.py:107]: Customers can still also use tokens in old format - this line will reveal the token.
Characters until first dot are usually the same in new format:  'dt0c01'

File: src/lib/fast_check.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def valid_dynatrace_scopes(token_metadata: dict):
 async def get_dynatrace_token_metadata(dt_session: ClientSession, context: LoggingContext, dynatrace_url: str, dynatrace_api_key: str, timeout: Optional[int] = 2) -> dict:
     try:
         response = await dt_session.post(
-            url=f"{dynatrace_url.rstrip('/')}/api/v1/tokens/lookup",
+            url=f"{dynatrace_url.rstrip('/')}/api/v2/apiTokens/lookup",
```
Comment (Dawid-Kaszubski) [inline src/lib/fast_check.py:79]: Does `v2/apiTokens` support tokens in old format?

Comment (dtwwolinski) [reply src/lib/fast_check.py:79]: It does. I've chosen but the output of v2 is more readable.

Comment (dtwwolinski) [reply src/lib/fast_check.py]: Reverted to v1 as the name of the token is the only needed thing.

File: src/lib/fast_check.py [hunk 3]
```diff
@@ -96,6 +96,36 @@ async def get_dynatrace_token_metadata(dt_session: ClientSession, context: Loggi
         return {}
 
 
+def format_dynatrace_access_key(dynatrace_access_key :str):
+    if len(dynatrace_access_key) >= 7:
+        if dynatrace_access_key.startswith("dt0c01.") and len(dynatrace_access_key) >= 14:
```
[the remark this answers is not in the document]
Comment (sq2gxo) [reply src/lib/fast_check.py:101]: we cannot assume the lenght, it is good to be defensive against IndexError

File: src/lib/fast_check.py [hunk 4]
```diff
@@ -168,8 +178,6 @@ async def execute(self) -> FastCheckResult:
             if all(result is not None for result in results):
                 ready_to_monitor.append(project_id)
 
-        await self._check_dynatrace(get_project_id_from_environment())
```
Comment (sq2gxo) [inline src/lib/fast_check.py:171]: I think that check was executed for K8S runtime and should stay here also

File: src/lib/fast_check.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (karolkieglerski) [reply src/lib/fast_check.py:101]: @sq2gxo didn't catch your answer

[the remark this answers is not in the document]
Comment (sq2gxo) [reply src/lib/fast_check.py:101]: ignore my answer, I didn't catch you question :)

**Review by sq2gxo — APPROVED** [review]

**Review by Dawid-Kaszubski — APPROVED** [review]

**Pull request merged**